ckanext/dcat/processors.py

```diff
@@ -330,6 +333,70 @@ def serialize_catalog(self, catalog_dict=None, dataset_dicts=None,
 
         return output
 
+    def _add_source_catalog(self, root_catalog_ref, dataset_dict, dataset_ref):
+        if not p.toolkit.asbool(config.get(DCAT_EXPOSE_SUBCATALOGS, False)):
+            return
+
+        def _get_from_extra(key):
+            for ex in dataset_dict.get('extras', []):
+                if ex['key'] == key:
+                    return ex['value']
+
+        source_uri = _get_from_extra('source_catalog_homepage')
+        if not source_uri:
+            return
+        
+        g = self.g
+        catalog_ref = URIRef(source_uri)
+
+        catalog_added = g.value(root_catalog_ref, DCT.hasPart) == catalog_ref
+
+        g.add((root_catalog_ref, DCT.hasPart, catalog_ref))
```

Comment:
Shouldn't these two be inside `if not catalog_added`?

```
g.add((root_catalog_ref, DCT.hasPart, catalog_ref))
g.add((catalog_ref, RDF.type, DCAT.Catalog))
```

Reply:
Sure. Not spotted before because it was simply adding the same node over and over.